R/utils.R

```diff
@@ -1,44 +1,41 @@
-#' Removes curly brackets from string
+#' Removes leading and trailing curly brackets from character
+#' string and removes indentation of remaining contents
 #'
 #' @description `r lifecycle::badge("stable")`
 #' @param x (`character`)\cr
-#'  string with code chunk
 #'
-#' @return The string without curly braces
+#' @return Character string without curly braces
 #' @keywords internal
-#'
 remove_enclosing_curly_braces <- function(x) {
   checkmate::assert_character(x)
   if (length(x) == 0) {
-    x
-  } else {
-    txt <- unlist(strsplit(x, "\n", fixed = TRUE))
-    if (grepl("^\\{[[:space:]]*", txt[1]) && grepl("^\\}[[:space:]]*", utils::tail(txt, 1))) {
-      txt2 <- txt[-c(1, length(txt))]
+    return(x)
+  }
+
+  open_bracket_then_spaces <- "^\\{[[:space:]]*"
+  close_bracket_then_spaces <- "^\\}[[:space:]]*"
+  blank_line <- "^[[:space:]]*$"
+  four_spaces_at_start_of_line <- "^[[:space:]]{4}"
 
-      n_spaces_indent <- vapply(txt2, function(txt_i) {
-        if (grepl("^[[:space:]]*$", txt_i)) {
-          NA_integer_
-        } else {
-          txt_i <- "    a <-   "
-          i <- which(diff(which(unlist(strsplit(txt_i, " ", fixed = TRUE)) == "")) != 1)[1]
+  split_text <- unlist(strsplit(x, "\n", fixed = TRUE))
 
-          if (length(i) == 1 && i > 0) {
-            i
-          } else {
-            NA_integer_
-          }
-        }
-      }, numeric(1), USE.NAMES = FALSE)
+  # if text begins with "{   \n" and ends with "\n}   "
+  if (grepl(open_bracket_then_spaces, utils::head(split_text, 1)) &&
+      grepl(close_bracket_then_spaces, utils::tail(split_text, 1))) {
+    # remove the first and last line
+    split_text <- split_text[-c(1, length(split_text))]
 
-      if (sum(!is.na(n_spaces_indent)) > 0) {
-        n_rm <- min(n_spaces_indent, na.rm = TRUE)
-        gsub(paste0("^[[:space:]]{", n_rm, "}"), "", txt2)
-      } else {
-        txt2
-      }
-    } else {
-      txt
+    # if any line is not blank then indent
+    if (!all(grepl(blank_line, split_text))) {
+      return(gsub(four_spaces_at_start_of_line, "", split_text))
     }
+  } else {
+    return(split_text)
   }
 }
+
+
+# converts vector of expressions to character
+format_expression <- function(code) {
```

Comment:
Note I have to leave `remove_enclosing_curly_braces` as a stand alone function as chunks use it.